source/_posts/2015-06-01-creating-and-publishing-a-ruby-gem.markdown

```diff
@@ -0,0 +1,345 @@
+
+	  spec.files        = `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
+	  spec.bindir       = "exe"
+	  spec.executables  = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
+	  spec.require_paths= ["lib"]
+
+	  spec.add_development_dependency "bundler", "~> 1.8"
+	  spec.add_development_dependency "rake", "~> 10.0"
+	end
+
+This file contains metadata about your gem and it can be populated directly, so here you can enter all the data such as name, description, licence... This file also contains information about what files should be packaged in your gem, as well as the load path to include the gem directory when the gem is first loaded. Most of these default settings will work for the majority of gems but you can always edit them if you want different behavior. At the bottom of the file add any gem dependencies that are required.
+
+The version number of the gem is set in `SlugConverter::VERSION` constant which is kept in a separate version.rb file, and you can change it there for every new version of your gem.
+
+	lib
+	  |--slug_converter
+  	  			|--version.rb
+
+A very important part of every gem is the README file, where you can describe how to install and use your gem, and the LICENCE file where you can define the terms and conditions under which your gem can be used.
+
+In the lib directory there is a file which has the same name as your gem (recommended), and that file will be loaded when someone requires your gem. If the gem you are writing is simple all the code can be placed in this single file, or in case of more complex gems all the other files from the lib directory are required in this file.
+
+There is also a Gemfile generated, but this file doesn't have to be managed directly since all it does is look in .gemspec for required dependencies and then loads them through bundler. All the dependencies required by the gem should be specified in the gemspec file.
+
+Another file that is generated by the bundler is Rakefile which just adds some gem tasks from bundler, and we can see those tasks with explanation by running 
+	rake -T
+	rake build    # Build slug_converter-0.0.1.gem into the pkg directory
+	rake install  # Build and install slug_converter-0.0.1.gem into system gems
+	rake release  # Create tag v0.0.1 and build and push slug_converter-0.1.0.gem to Rubygems
+
+
+## Writing tests
+
+If you are following the principles of Test Driven Development you will probably like to start by writing tests for you gem, for that purpose I would suggest using RSpec. 
+To do that you will need to add rspec as a development dependency to you gemspec file:
+	spec.add_development_dependency 'rspec'
+
+It was mentioned in the beginning that, when running bundle gem for the first time bundler will asks if you would like to generate test files for your gem and to choose if you want to use rspec or minitest. If you answer with yes, and choose rspec, bundler will generate a spec directory with two files: 
```

Comment:
@codalicious why so passive? :smiley: Maybe use something like: `as mentioned earlier` or `as I mentioned in the beginning.